### PR TITLE
Optimize float values in the data file

### PR DESCRIPTION
**Pull request by mathiasb**
Calculate percentage values directly in the JSON file for the Electric car stats. Add command line parameter to climate_data_calculation.py so that with the parameter -n <number> the number of decimals can be specified for all floating point numbers in the file. The file is then named differently from the unrounded file (as output by the script).

### CODE REVIEW

**Comment by Greenheart** [comment]
Wow! Thank you Mathias - so nice to see these new features and improved structure. Looking forward to test and give a proper review, but from a quick look this seems really good! :)

File: data/solutions/cars/electric_car_change_rate.py [hunk 1]
```diff
@@ -13,13 +13,13 @@ def get_electric_car_change_rate(df):
...
 
-    df_raw_cars["electricCarChangePercent"] = df_raw_cars[
+    df_raw_cars["electricCarChangePercent"] = 100.0 * df_raw_cars[
         "Procentenheter förändring av andel laddbara bilar 2015-2022"
     ]
     
     years = [2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022]
     df_raw_cars["electricCarChangeYearly"] = df_raw_cars.apply(
-        lambda x: {year: x.loc[year] for year in years},
+        lambda x: {year: (100.0 * x.loc[year]) for year in years},
```
Comment (Greenheart) [inline data/solutions/cars/electric_car_change_rate.py:22]: We likely need to make frontend changes to expect percentage values.

**Comment by elvbom** [comment]
Tack för denna fina PR, bra idé och bra jobb! Vi tittar på den efter helgen!

**Comment by Greenheart** [comment]
WARNING: Do not merge this PR until we have reverted the latest commits that included changes from the `main` branch. This would cause severe merge conflicts in all other branches.

We should probably 1) revert to commit https://github.com/Klimatbyran/klimatkollen/pull/530/commits/ca25ec1520ef568d5013f06a48e2af916cecd647 and then 2) run something like the following:

TL;DR: we need to merge staging into feature branches. `main` is our production branch. @elvbom Maybe we should rename main to production to make it super obvious?

```bash
# Assuming the current branch is `climate-data-value-trunc-feature`
git pull [REMOTE] staging # (where [REMOTE] is replaced with origin/upstream depending on which one is the Klimatbyran/klimatkollen repo)
git merge staging # merge latest staging into this feature branch.
# Then resolve the conflicts and commit.
git push # Finally git push to update this PR
```

**Comment by mathiasb** [comment]
Hej! Det blev enklare att ta ett nytt tag på att lösa detta, så det kommer två olika PR. En för electric vehicle procent-beräkningen, och en PR för att sätta hur många decimaler man vill ha. Denna PR kan stängas därmed.

**Comment by Greenheart** [comment]
Sorry, my last message was a bit aggressive. Nothing against your contributions here - I just wanted to highlight that a similar situation caused problems recently 😊

I also know that @elvbom is working on updates related to what you have started here. Hopefully we can get this merged soon to reduce the data size.

**Comment by mathiasb** [comment]
> Sorry, my last message was a bit aggressive. Nothing against your contributions here - I just wanted to highlight that a similar situation with the merge conflicts caused problems recently 😊
> 
> I also know that @elvbom is working on updates related to what you have started here. Hopefully we can get this merged soon to reduce the data size.

not perceived that way at all! it's great with orderly routines and clear expectations. There are now new PR's (#612 and #613 ), so this one can be closed.

**Comment by elvbom** [comment]
> > Sorry, my last message was a bit aggressive. Nothing against your contributions here - I just wanted to highlight that a similar situation with the merge conflicts caused problems recently 😊
> > I also know that @elvbom is working on updates related to what you have started here. Hopefully we can get this merged soon to reduce the data size.
> 
> not perceived that way at all! it's great with orderly routines and clear expectations. There are now new PR's (#612 and #613 ), so this one can be closed.

Thank you for letting us know and great job! I'll look at the other PRs instead and close this one.
